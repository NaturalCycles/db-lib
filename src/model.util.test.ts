import { deepFreeze, mockTime } from '@naturalcycles/test-lib'
import { assignIdCreatedUpdated } from './model.util'

beforeEach(() => {
  jest.restoreAllMocks()
  mockTime()
  jest.spyOn(require('@naturalcycles/nodejs-lib'), 'stringId').mockImplementation(() => 'someId')
})

test('assignIdCreatedUpdated', () => {
  const o = {
    id: '123',
    updated: 123,
    created: 123,
  }

  deepFreeze(o)

  // Should preserve
  expect(assignIdCreatedUpdated(o, true)).toEqual(o)
  expect(assignIdCreatedUpdated(o)).toMatchSnapshot()
})
