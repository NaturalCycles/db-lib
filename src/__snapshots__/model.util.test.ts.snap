// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assignIdCreatedUpdated 1`] = `
Object {
  "created": 123,
  "id": "123",
  "updated": 1529539200,
}
`;
